src/main.rs

```diff
@@ -104,6 +104,13 @@ fn main() -> Result<()> {
                 config.rename_context(ctx_id.unwrap(), new_ctx)?;
                 config.write(config_path)?;
             }
+            Context_subcommands::set_default_algo => {
+                let algo = c.unwrap().value_of(&Parameters::algo).unwrap().to_string();
```

Comment:
Since you defined the `SignAlgo` enum, you could parse the value to a enum variant. It would be cleaner than handling a String all around :)